tests/checker/good/lib_typing.scilla

```diff
@@ -0,0 +1,10 @@
+scilla_version 0
+
+library LibTyping
+
+let x : Uint32 = Uint32 0
+let y = "Test String"
+let z = Uint128 5
```

Comment:
I would like some tests of complex types (ADTs and functions in particular), as well as some negative tests (i.e., test where the type annotation is incorrect).

In particular, I would like to see a test where the type of the function is `tfun 'A => Uint128 -> Uint128`, but where the type annotation is `'A => 'A => 'A`.

Also, if we decide to require that the type annotation is identical (up to alpha-renaming), then I would like to see the test case where the type annotation is less general than the actual type as well.

Reply:
@jjcnn I've added the required test. Have a look when you can.

Reply:
Thanks @vaivaswatha!